src/common/telemetry/inc/Telemetry.h

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef TELEMETRY_H
+#define TELEMETRY_H
+
+#include <Logging.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// Opens the static telemetry logger instance
+// Returns 0 on success, non-zero on failure
+int OSConfigTelemetryOpen(OsConfigLogHandle log);
+
+// Closes the static telemetry logger instance
+// Returns 0 on success, non-zero on failure
+int OSConfigTelemetryClose(void);
+
+// eventName: Name of the event to log
+// keyValuePairs: Array of key-value pair strings (must be even number of elements)
+// pairCount: Number of key-value pairs (keyValuePairs array size / 2)
+// Returns 0 on success, non-zero on failure
+int OSConfigTelemetryLogEvent(const char* eventName, const char** keyValuePairs, int pairCount);
+
+// Set the binary directory path for telemetry executable on the static instance
+// directory: Path to the directory containing the telemetry binary
+// Returns 0 on success, non-zero on failure
+int OSConfigTelemetrySetBinaryDirectory(const char* directory);
+
+// Get the directory of the current module
+// Returns pointer to the directory string, or NULL on failure
+const char* OSConfigTelemetryGetModuleDirectory();
```

Comment:
Can we move this interface as internal into the C++ telemetry client and keep for the NRP client just one .h file that declares macros to write telemetry event lines to the telemetry log?

Reply:
Basically, to better decouple the client (such as the NRP) from the Telemetry Client. And in the process of this, to also make as much of the new Telemetry code C++.